Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts

```diff
@@ -268,21 +301,165 @@ namespace gdjs {
       );
     }
 
-    updateXPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.x = this._object.x;
-      } else {
+    invalidateBounds() {
+      this._object.invalidateBounds();
+      this._positionXIsUpToDate = false;
+      this._positionYIsUpToDate = false;
+    }
+
+    updatePreRender(): void {
+      this.updatePositionIfNeeded();
+    }
+
+    updatePositionX(): void {
+      if (this._object._absoluteCoordinates) {
+        this._graphics.pivot.x = 0;
         this._graphics.position.x = 0;
+      } else {
+        // Make the drawing rotate around the anchor.
+        this._graphics.pivot.x = this._object.getRotationAnchorX();
+        // Multiply with the scale to have the scale anchor
+        // at the object position instead of the center.
+        this._graphics.position.x =
+          this._object.x +
+          this._graphics.pivot.x * Math.abs(this._graphics.scale.x);
       }
     }
 
-    updateYPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.y = this._object.y;
-      } else {
+    updatePositionY(): void {
+      if (this._object._absoluteCoordinates) {
```

Comment:
maybe a lack of knowledge from my end, so apologies! 
But, what's the added value of allowing the object to have absolute coordinates?

Also, should this be called `useAbsoluteCoordinates` to make it clear this is a boolean and not actual coordinates?

Reply:
This is an existing property that allows user to draw directly in the scene basis without having every ShapePainter instances stacked at (0; 0) in the instance editor.

Yes, this name is less confusing.

Reply:
Yeah the naming is a bit confusing - though it was already there so nothing blocking for this PR.